### ISSUE TITLE
Invalid processing regexp while searching words including "("

### ISSUE THREAD

**Issue by seagle0128** [dajva/rg.el#19]
`GNU Emacs 25.2 on macOS 10.12.6, with [configurations](https://github.com/seagle0128/.emacs.d)

When searching `(lambda (` via `rg`, return errors.

```elisp
-*- mode: rg; default-directory: "~/.emacs.d/lisp/" -*-
rg started at Sat Sep  9 02:20:28

rg --color always --colors match:fg:red -n --type elisp --column --heading -i -e \(lambda\ \( .
Error parsing regex near 'mbda (' at character offset 8: Unclosed parenthesis.

rg exited abnormally with code 1 at Sat Sep  9 02:20:28
```

if change to `\(lambda \(`, `rg` works well. But `(lambda (` works fine with `ag` and `rgrep`. And I think `ag.el` handle it with `search the words at point` gracefully.

**Good**
```elisp
rg --color always --colors match:fg:red -n --type elisp --column --heading -i -e \\\(lambda\ \\\( .
```
```elisp
ag --literal --group --line-number --column --color --color-match 30\;43 --color-path 1\;32 --smart-case --stats -- \(lambda\ \( .
```

Please take a look into this issue. Thanks!

**Comment by dajva** [comment]
I already had a fix for `rg-dwim` locally that I have pushed now. See if it solves your use case.
Thanks for the report.

**Comment by seagle0128** [comment]
Thanks for the quick fixing, @dajva. I will check once it''s merged in mepla.

**Comment by seagle0128** [comment]
`rg-dwim` works well. Suggest to support in `rg` as well. Thanks!

**Comment by dajva** [comment]
The `rg` defun is a bit of another story since the input is actually supposed to be a regexp that ripgrep's regex engine understands. In your example the `(` and `)` are for grouping sub expressions and needs to be escaped to be litteral. `rgrep` actually has the same behavior but parenthesis seems to have no special meaning so this particular example works.

So if I understand you correctly you want a way to search with a litteral string in general instead of a regexp?
I guess the problem with the current approach is that if you want to change the search term in the results buffer with `r` you will then have to edit a regexp, which would be confusing. It could be a short term solution though since it's trivial to implement.

I need to think a bit about this but my first thought is to have a separate entry point, `rg-literal`, that sets the `--literal` flag and in the results buffer you will have the opportunity to toggle this flag if needed. An alternative is to add this flag to `rg-command-line-flags` but then I just broke this for `rg-dwim`. :-/

Any thoughs? How does `ag` handle this?

**Comment by seagle0128** [comment]
I think you are right. I did some tests with `ag` and `rg`.

For example, I want to search `(lambda (` in the project.
`ag` will convert the string into this command:
```elisp
ag --literal --group --line-number --column --color --color-match 30\;43 --color-path 1\;32 --smart-case --stats -- \(lambda\ \( .
```
`rg` will produce this:
```elisp
rg --color always --colors match:fg:red -n --type elisp --column --heading -i -e \(lambda\ \( .
```
with this error:
```
Error parsing regex near 'mbda (' at character offset 8: Unclosed parenthesis.
```
I tried to add `-F` or `--fixed-string` flag which means "Treat the pattern as a literal string instead of a regular expression". 
```elisp
rg --fixed-string --color always --colors match:fg:red -n --type elisp --column --heading -i -e \(lambda\ \( .
```
It worked well with no errors.

Then I did some research for the source codes of `ag.el`. I realized the `ag` command uses `--literal` flag by default, and `ag-regexp` is for the regexp search. It does make sense for the most of people, at least for me.

Of cource `rg` does great with the regexp by default, but a little bit inconvenient, because at the most of time the people just want to search with some fixed strings. Thinking of this, how about adding `rg-regexp` command to handle regexp searches and making `rg` to handle string searches? I think it would be simplify the implementation of `rg-dwim` as well.

These are just my thoughts. For now I set `rg-command-line-flags` and it works for me.

Appreciate to your great work and patience!
